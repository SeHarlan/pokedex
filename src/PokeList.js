import React, { Component } from 'react';
import PokeData from './mockData';
import request from 'superagent';

import PokeItem from './PokeItem.js';

export default class App extends Component {
    state = { pokeDex: [] }

    async componentDidMount() {
      const data = await request.get('https://alchemy-pokedex.herokuapp.com/api/pokedex')
      this.setState({ pokeDex: data.body.results})
    }


    render() {
        const pokeNodes = this.state.pokeDex
            .map(item => <PokeItem item={item} key={item._id} />)
      return <section>
          <ul>
            {pokeNodes}

          </ul>
      </section>
    }
  }