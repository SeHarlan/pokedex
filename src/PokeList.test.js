import React from "react";
import renderer from "react-test-renderer";
import PokeList from './PokeList.js';
import mockData from '../src/mockData.js'

test('renders pokelist.js correctly', () => {
  const tree = renderer
    .create(<PokeList pokemon={mockData}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})