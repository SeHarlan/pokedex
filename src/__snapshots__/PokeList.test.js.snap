// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders pokelist.js correctly 1`] = `
<section>
  <ul>
    <li
      className="poke-item"
      style={
        Object {
          "borderColor": "#78C850",
        }
      }
    >
      <h2>
        venusaur
      </h2>
      <div>
        <span
          className="type"
          style={
            Object {
              "backgroundColor": "#78C850",
            }
          }
        >
          grass
        </span>
        <span>
          HP: 
          80
        </span>
      </div>
      <img
        alt="venusaur"
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/003.png"
      />
      <div>
        <span>
          A: 
          82
        </span>
        <span>
          D: 
          83
        </span>
      </div>
    </li>
    <li
      className="poke-item"
      style={
        Object {
          "borderColor": "#78C850",
        }
      }
    >
      <h2>
        venusaur
      </h2>
      <div>
        <span
          className="type"
          style={
            Object {
              "backgroundColor": "#78C850",
            }
          }
        >
          grass
        </span>
        <span>
          HP: 
          80
        </span>
      </div>
      <img
        alt="venusaur"
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/003.png"
      />
      <div>
        <span>
          A: 
          82
        </span>
        <span>
          D: 
          83
        </span>
      </div>
    </li>
    <li
      className="poke-item"
      style={
        Object {
          "borderColor": "#78C850",
        }
      }
    >
      <h2>
        bulbasaur
      </h2>
      <div>
        <span
          className="type"
          style={
            Object {
              "backgroundColor": "#78C850",
            }
          }
        >
          grass
        </span>
        <span>
          HP: 
          45
        </span>
      </div>
      <img
        alt="bulbasaur"
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/001.png"
      />
      <div>
        <span>
          A: 
          49
        </span>
        <span>
          D: 
          49
        </span>
      </div>
    </li>
    <li
      className="poke-item"
      style={
        Object {
          "borderColor": "#78C850",
        }
      }
    >
      <h2>
        venusaur-mega
      </h2>
      <div>
        <span
          className="type"
          style={
            Object {
              "backgroundColor": "#78C850",
            }
          }
        >
          grass
        </span>
        <span>
          HP: 
          80
        </span>
      </div>
      <img
        alt="venusaur-mega"
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/003_f2.png"
      />
      <div>
        <span>
          A: 
          100
        </span>
        <span>
          D: 
          123
        </span>
      </div>
    </li>
  </ul>
</section>
`;
